sql: schema changer crashes when table not configured with zone cfg

```
E190320 15:54:50.350885 207 util/log/crash_reporting.go:202  [n1] a panic has occurred!
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x213f379]

goroutine 207 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc0004b4480, 0x39d8080, 0xc00077a660)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:185 +0x11f
panic(0x2e80ca0, 0x56971d0)
        /usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChangeManager).Start.func1.3(0xc00dcab270, 0x5, 0x8, 0xc00d928580, 0x575, 0x580, 0x158db5c1b716ce27, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:1740 +0xa79
github.com/cockroachdb/cockroach/pkg/gossip.(*SystemConfigDeltaFilter).ForModified(0xc00d49dac0, 0xc00d250e10, 0xc00d49dc28)
        /go/src/github.com/cockroachdb/cockroach/pkg/gossip/util.go:90 +0x313
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChangeManager).Start.func1(0x39d8080, 0xc00077a660)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:1702 +0x91b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc00026e610, 0xc0004b4480, 0xc0003a1ec0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:200 +0xe1
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:193 +0xa8
```

--- a/pkg/sql/schema_changer.go
+++ b/pkg/sql/schema_changer.go
@@ -1572,6 +1572,7 @@ func (s *SchemaChangeManager) Start(stopper *stop.Stopper) {
 		if s.testingKnobs.AsyncExecQuickly {
 			delay = 20 * time.Millisecond
 		}
+		defTTL := config.DefaultZoneConfig().GC.TTLSeconds
 
 		execOneSchemaChange := func(schemaChangers map[sqlbase.ID]SchemaChanger) {
 			for tableID, sc := range schemaChangers {
@@ -1631,8 +1632,12 @@ func (s *SchemaChangeManager) Start(stopper *stop.Stopper) {
 					for id, sc := range s.forGC {
 						zoneCfg, placeholder, _, err := ZoneConfigHook(cfg, uint32(id))
 						if err != nil {
-							log.Errorf(ctx, "no zone config for desc: %d", id)
+							log.Errorf(ctx, "zone config for desc: %d, err = %+v", id, err)
 							return
+						}
+						if zoneCfg == nil {
+							// Do nothing, use the old zone config's TTL.
+							continue
 						}
 						if placeholder == nil {
 							placeholder = zoneCfg
@@ -1727,22 +1732,25 @@ func (s *SchemaChangeManager) Start(stopper *stop.Stopper) {
 						if len(table.GCMutations) > 0 {
 							zoneCfg, placeholder, _, err := ZoneConfigHook(cfg, uint32(table.ID))
 							if err != nil {
-								log.Errorf(ctx, "no zone config for desc: %d", table.ID)
+								log.Errorf(ctx, "zone config for desc: %d, err = %+v", table.ID, err)
 								return
 							}
+
 							if placeholder == nil {
 								placeholder = zoneCfg
 							}
 
 							for _, m := range table.GCMutations {
-								ttlSeconds := zoneCfg.GC.TTLSeconds
-								if subzone := placeholder.GetSubzone(uint32(m.IndexID), ""); subzone != nil {
-									ttlSeconds = subzone.Config.GC.TTLSeconds
+								// Initialize TTL without a zone config in case it's not present.
+								ttlSeconds := defTTL
+								if zoneCfg != nil {
+									ttlSeconds = zoneCfg.GC.TTLSeconds
+									if subzone := placeholder.GetSubzone(uint32(m.IndexID), ""); subzone != nil {
+										ttlSeconds = subzone.Config.GC.TTLSeconds
+									}
 								}
+								deadline := m.DropTime + int64(delay) + int64(ttlSeconds)*time.Second.Nanoseconds()
 
-								deadline := m.DropTime +
-									int64(ttlSeconds)*time.Second.Nanoseconds() +
-									int64(delay)
 								dropped := droppedIndex{m.IndexID, m.DropTime, deadline}
 								if minDeadline == 0 || deadline < minDeadline {
 									minDeadline = deadline
@@ -1776,12 +1784,17 @@ func (s *SchemaChangeManager) Start(stopper *stop.Stopper) {
 								schemaChanger.dropTime = table.DropTime
 								zoneCfg, _, _, err := ZoneConfigHook(cfg, uint32(table.ID))
 								if err != nil {
-									log.Errorf(ctx, "no zone config for desc: %d", table.ID)
+									log.Errorf(ctx, "zone config for desc: %d, err: %+v", table.ID, err)
 									return
 								}
-								deadline := table.DropTime +
-									int64(zoneCfg.GC.TTLSeconds)*time.Second.Nanoseconds() +
-									int64(delay)
+
+								// Initialize deadline without a zone config in case it's not present.
+								deadline := table.DropTime + int64(delay)
+								if zoneCfg != nil {
+									deadline += int64(zoneCfg.GC.TTLSeconds) * time.Second.Nanoseconds()
+								} else {
+									deadline += int64(defTTL) * time.Second.Nanoseconds()
+								}
 								if minDeadline == 0 || deadline < minDeadline {
 									minDeadline = deadline
 								}

--- a/pkg/sql/zone_config.go
+++ b/pkg/sql/zone_config.go
@@ -149,7 +149,8 @@ func completeZoneConfig(
 
 // ZoneConfigHook returns the zone config for the object with id using the
 // cached system config. If keySuffix is within a subzone, the subzone's config
-// is returned instead.
+// is returned instead. The bool is set to true when the value returned is
+// cached.
 func ZoneConfigHook(
 	cfg *config.SystemConfig, id uint32,
 ) (*config.ZoneConfig, *config.ZoneConfig, bool, error) {

